pyretailscience/segmentation/segstats.py

```diff
@@ -207,18 +253,28 @@ def _calc_seg_stats(
         if calc_rollup:
             rollup_metrics = []
 
-            # Generate all prefixes of segment_col (except the empty prefix which is the grand total)
-            # and excluding the full segment_col list which is already calculated
-            for i in range(1, len(segment_col)):
-                prefix = segment_col[:i]
-
-                # Group by the prefix and aggregate
-                rollup_result = data.group_by(prefix).aggregate(**aggs)
+            # Configuration for rollups
+            rollup_configs = [
+                # Prefix rollups: always include when calc_rollup=True
+                *[(segment_col[:i], segment_col[i:], rollup_value[i:]) for i in range(1, len(segment_col))],
+            ]
```

Comment:
This can be simplified,
```suggestion
            rollup_configs = [
                # Prefix rollups: always include when calc_rollup=True
                (segment_col[:i], segment_col[i:], rollup_value[i:]) for i in range(1, len(segment_col)).
            ]
```